homeassistant/components/deconz/light.py

```diff
@@ -67,8 +72,13 @@ def async_add_group(groups):
         entities = []
 
         for group in groups:
-            if group.lights:
-                entities.append(DeconzGroup(group, gateway))
+            if not group.lights:
+                continue
+
+            known_groups = list(gateway.entities[DOMAIN])
```

Comment:
Why do we copy to a list instead of keeping it as a set?

Reply:
True, the focus is on the copy, since creating the object will register it in the domain so we need the snapshot prior to creating it.

Reply:
Fixed in #40640